client/my-sites/exporter/post-type-options.jsx

```diff
@@ -0,0 +1,174 @@
+/**
+ * External dependencies
+ */
+import React, { PropTypes } from 'react';
+import PureRenderMixin from 'react-pure-render/mixin';
+import { connect } from 'react-redux';
+import moment from 'moment';
+
+/**
+ * Internal dependencies
+ */
+import FormRadio from 'components/forms/form-radio';
+import Select from './select';
+import Label from 'components/forms/form-label';
+
+import { setPostType } from 'state/site-settings/exporter/actions';
+import {
+	getPostTypeOptions,
+	getSelectedPostType,
+} from 'state/site-settings/exporter/selectors';
+
+const mapStateToProps = ( state, ownProps ) => {
+	const siteId = state.ui.selectedSiteId;
+	const options = getPostTypeOptions( state, siteId, ownProps.postType );
+
+	return {
+		options,
+
+		// Show placeholders when options are not available
+		shouldShowPlaceholders: ! options,
+
+		// Disable options when this post type is not selected
+		isEnabled: getSelectedPostType( state ) === ownProps.postType,
+	}
+}
+
+const mapDispatchToProps = ( dispatch, ownProps ) => {
+	return {
+		onSelect: () => dispatch( setPostType( ownProps.postType ) )
+	}
+}
+
+/**
+ * Displays a list of select menus with a checkbox legend
+ *
+ * Displays a field group with a checkbox legend and optionally
+ * a list of select menus, or a description to appear beneath the
+ * legend.
+ */
+
+const PostTypeOptions = React.createClass( {
+	displayName: 'PostTypeOptions',
+
+	mixins: [ PureRenderMixin ],
+
+	propTypes: {
+		onSelect: PropTypes.func,
+
+		legend: PropTypes.string.isRequired,
+	},
+
+	renderPlaceholders() {
+		return (
+			<div className="exporter__option-fieldset-fields">
+				<div className="exporter__placeholder-text">
+					{ this.translate( 'Loading options…' ) }
+				</div>
+			</div>
+		);
+	},
+
+	renderFields() {
+		const { options } = this.props;
+		const commonProps = {
+			disabled: ! this.props.isEnabled
+		};
+
```

Comment:
Just a suggestion:

I see a lot of repeated `value`/`label` and also function declarations that are only used once. Maybe you could use one function and simplify all of the other mappings like this
```js
const postTypeOption = ( value, label ) => { value, label };
const mappedAuthors = options.authors && options.authors.map( author => postTypeOption( author.ID, author.name ) );
const mappedStatuses = options.statuses && options.statuses.map( status => postTypeOption( status.name, status.label ) );
...
```

And then...

```js
...
{ mappedAuthors && <Select defaultLabel="Status…" options={ mappedAuthors } { ...commonProps } /> }
...
```

You could probably take it even further to eliminate the repeating with

```js
const selectProps = [];
options.authors && selectProps.push( {
    defaultLabel: this.translate( 'Author…' ),
    options: options.authors.map( author => postTypeOption( author.ID, author.name ) )
} );
...
```

then

```js
...
{ selectProps.map( props => <Select { ...props } disabled={ disabled } /> }
...
```

This is just a suggestion and not trying to rewrite :smile: